packages/core/docs/general/key-concepts.md

```diff
@@ -2,122 +2,61 @@
-
-Middleware by default is a part of your Nuxt application, but it can be detached and used as a separate node application.
-
-You can read more about Vue Storefront Middleware on the [Server Middleware](/advanced/server-middleware) page.
-
-## Integrations and extendibility
+You can read more about it on the [Configuration](/guide/configuration.html) page.
 
-All 3rd party integrations (eCommerce, CMS, Search, etc.) can be added and configured in `middleware.config.js`:
+## Reusability
 
-```js
-//middleware.config.js
-module.exports = {
-  integrations: {
-    // ...
-  }
-}
-```
+To provide all necessary tools to create customizable and easy to maintain applications, Vue
+Storefront offers composables - functions that implement most of business logic of the application.
 
-Each integration has an `extensions` field. You can use extensions to:
-- override existing API methods of a certain integration;
-- add new API methods;
-- change method parameters before they are called;
-- react to a method being called;
+**Composables are the main public API of Vue Storefront** and in many cases the only API you'll work
+with. You can treat each composable as an independent micro-application. They manage their state,
+handle SSR and in most cases don't interact with each other.
 
 ```js
-{
-  name: 'my-extension',
-  extendApiMethods: {
-    // will override default getProduct
-    getProduct: async () => { /* ... */ }
-    // will add new method to the integration 
-    doSomethingMore: async () => { /* ... */}
-  },
-  hooks: (req, res) => {
-    return {
-      beforeCreate: ({ configuration }) => configuration,
-      afterCreate: ({ configuration }) => configuration,
-      beforeCall: ({ configuration, callName, args }) => args,
-      afterCall: ({ configuration, callName, args, response }) => response
-    }
-  }
-}
+const { search, product, loading, error } = useProduct();
 ```
 
-Sometimes you just need to call a specific API client method without using a composable. Luckily you have access to all API methods of the registered integrations through `useVSFContext` composable. 
+You can read more about them on the [Composables](/guide/composables.html) page.
 
-The composable returns a list of tags representing your integrations (it's usually either a name of the integration like `$storyblok` or an acronym for longer names like `$ct` for commercetools).
+## Extendability and custom integrations
 
-```js
-const { $ct } = useVsfContext()
-```
+Though some 3rd parties integrations are already covered and available in Vue Storefront repositories 
+and registries, Vue Storefront enables developers to create their own integrations with any platform
+by extending Server Middleware functionality - one of the key components that works as a bridge 
+between frontend and the backends.
```

Comment:
```suggestion
Although some integrations are already provided by the Vue Storefront team, we enable developers to create their own integrations with any platform
by extending the Server Middleware. It's the key component that works as a bridge
between the frontend and your backend of choice.
```
`Though`, `thought`, `tough` -I like to avoid these words for obvious reasons :D